packaging/os/yum.py

```diff
@@ -556,33 +569,34 @@ def install(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos):
                 res['msg'] += "No Package file matching '%s' found on system" % spec
                 module.fail_json(**res)
 
-            nvra = local_nvra(module, spec)
+            pkg_name = local_name(module, spec)
```

Comment:
We have to pass just the package name here and not the full package version and arch, so that we can check if the package is installed. We don't care if the versions don't match, only if the package is installed.